src/http/ngx_http_upstream_round_robin.c

```diff
@@ -510,6 +678,10 @@ ngx_http_upstream_get_round_robin_peer(ngx_peer_connection_t *pc, void *data)
         ngx_http_upstream_rr_peers_wlock(peers);
```

Comment:
Like above, this can be simplified to avoid locking. Just call `ngx_http_upstream_get_round_robin_peer()` recursively and return its result